### PR TITLE
ESLint/JSCS critical changes

### PR DESCRIPTION
**Pull request by brettz9**
Hi,

If you don't want all of the JSLint-inspired changes in #5 (or even if you do), I would at least suggest accepting this PR.

In this PR, I began pretty much with the strictest ESLint/JSCS settings possible (at least bearable ones), and then reenabled the settings to allow for all stylistic choices you have made.

I then identified a number of problems which are all fixed as per below, with the exception of one block which refers to `this.changes` and `diff.changes` which don't appear to be set elsewhere in the code and are at: https://github.com/metagriffin/indexeddb-js/blob/master/src/indexeddb-js.js#L521

Here are the changes I did make:
- Remove skeleton of deprecated `setVersion`;
- README: `deleteDatabase` is now implemented;
- ESLint (use strict optimization, remove unused function and variable,
  avoid setting global unnecessarily, fix naming of variable from `req` to `request`, fix indent and trailing spaces, remove function from loop in favor of creating safe closure within the loop,
  consistent semi-colon usage and trailing space removal);
- JSCS (inconsistent indent within `deleteDatabase`)

### CODE REVIEW

File: src/indexeddb-js.js [hunk 1]
```diff
@@ -18,7 +18,7 @@
 if ( typeof(define) !== 'function')
   var define = require('amdefine')(module);
 
-define(['underscore'], function(_) {
+define(['underscore'], function(_) {'use strict';
```
Comment (metagriffin) [inline src/indexeddb-js.js]: can you move this to a line by itself? i prefer that -- it makes it clearer that it is not part of the import definition.

File: src/indexeddb-js.js [hunk 2]
```diff
@@ -922,17 +912,6 @@ define(['underscore'], function(_) {
     };
 
     //-------------------------------------------------------------------------
-    this.setVersion = function(version) {
```
Comment (metagriffin) [inline src/indexeddb-js.js]: can you please not delete this? it is for backwards compatibility with the old indexedDB spec before `setVersion` was replaced by `onupgradeneeded`. even better, make the error message indicate that it has been deprecated in favor of onupgradeneeded :)

File: test/test-indexeddb-js.spec.js [hunk 3]
```diff
@@ -15,7 +17,7 @@ define([
   'underscore',
   'sqlite3',
   '../src/indexeddb-js'
-], function(_, sqlite3, indexeddbjs) {
+], function(_, sqlite3, indexeddbjs) {'use strict';
```
Comment (metagriffin) [inline test/test-indexeddb-js.spec.js]: same thing here -- please put it on a line by itself.

**Comment by brettz9** [comment]
Hi,

Sorry for the delayed response. Just getting back into the project (at least for now).

I've made the suggested changes, and also added a `contributors` property to `package.json`.

**Comment by brettz9** [comment]
Btw, as far as the deprecation error, I'm not sure what the significance of the numbers in those errors are (here "10"), but I've carried it over anyways.

**Comment by metagriffin** [comment]
awesome, thanks!

re error numbers: that's just an old habit of mine... back in the c++ days, that was an easy way for me uniquely identify an error when clients gave me the error code. the numbering has gaps so that i could later add more errors between two existing errors. very effective back then, now better done with stack traces...

**Pull request merged**